cmd/kubeadm/app/images/images.go

```diff
@@ -104,8 +103,16 @@ func GetControlPlaneImages(cfg *kubeadmapi.ClusterConfiguration) []string {
 		images = append(images, GetEtcdImage(cfg))
 	}
 
+	// If the add-on is not explicitly configured (for backward compatibility)
+	// or set as true, then download the image for the add-on.
+	// Append the kube-proxy images
+	if flag, exist := cfg.AddOns["kube-proxy"]; !exist || flag {
+		images = append(images, GetKubernetesImage(constants.KubeProxy, cfg))
+	}
 	// Append the appropriate DNS images
-	images = append(images, GetDNSImage(cfg))
+	if flag, exist := cfg.AddOns["coredns"]; !exist || flag {
+		images = append(images, GetDNSImage(cfg))
+	}
```

Comment:
do we need to tell the user with a high verbosity `klog.V(x).Infof` that the pull of certain addons images are skipped?

Reply:
Done.